OnBaseDocsApi/Models/Profiles.cs

```diff
@@ -1,26 +1,15 @@
 using System.IO;
 using System.Collections.Generic;
-using Newtonsoft.Json.Linq;
 
 namespace OnBaseDocsApi.Models
 {
     public class Profiles
     {
         private readonly Dictionary<string, Credentials> _profiles;
 
-        public Profiles(string path)
+        public Profiles(YamlDotNet.Serialization.IDeserializer deserializer, string path)
         {
-            _profiles = new Dictionary<string, Credentials>();
-            var config = JObject.Parse(File.ReadAllText(path));
-
-            foreach (var prop in (config as JObject).Properties())
-            {
-                _profiles[prop.Name] = new Credentials
-                {
-                    Username = prop.Value.Value<string>("username"),
-                    Password = prop.Value.Value<string>("password")
-                };
-            }
+            _profiles = deserializer.Deserialize<Dictionary<string, Credentials>>(File.ReadAllText(path));
```

Comment:
More the Deserialize to the caller (Globals). Just have a parameter of Dictionary<string, Credentials> to the constructor.